lib/SimpleSAML/Locale/Translate.php

```diff
@@ -81,7 +81,8 @@ public static function translateSingularGettext(?string $original): string
         // This may happen if you forget to set a variable and then run undefinedVar through the trans-filter
         $original = $original ?? 'undefined variable';
 
-        $text = BaseTranslator::$current->gettext($original);
+        $loc = new Localization(Configuration::getInstance());
```

Comment:
I believe this is the main thing that breaks it. Everytime a string is translated, an new Localization object is initialized so all things done to the Localization object associated to the template are not used, 'added' domains are gone etc.

The translation is done on a 'default' initialized Localization object (plus: for every string that is translated, the main PO file is loaded and parsed...)

Not sure yet what a better solution is but pretty sure that this is at least part of the cause.